client/lib/user/user.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+import { Language } from '../../languages';
+import { GravatarOptions } from './gravatar';
+import { URL } from '../../types';
+
+/**
+ * Takes an optional callback function. If no callback is passed in, it will return a promise.
+ **/
+export interface RequestHandler {
+	( fn: ( err: unknown, res: unknown, headers: unknown ) => unknown ): unknown;
+	(): Promise< unknown >;
+}
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: () => void;
+	get: () => UserData;
+	fetch: () => Promise< void >;
+	handleFetchFailure: ( error: Error ) => void;
+	handleFetchSuccess: ( userdata: UserData ) => void;
+	getLanguage: () => Language | void;
+	getAvatarUrl: ( options: GravatarOptions ) => URL;
+	clear: () => Promise< void > | void;
+	sendVerificationEmail: RequestHandler;
+	set: ( attributes: UserData ) => boolean;
+	decrementSiteCount: () => void;
+	incrementSiteCount: () => boolean | void;
+	verificationPollerCallback: ( signal: unknown ) => void;
```

Comment:
specifying `unknown` here will force us to pass a parameter and won't type-check it. we can replace this with an optional `boolean` which may be called from the `localStorage` event handler

```ts
verificationPollerCallback: (signal?: true) => void;
```

Yes indeed; we don't have to specify `boolean` even because `true` is the only value `signal` is ever designed to inhabit.